pci/src/vfio.rs

```diff
@@ -584,15 +589,15 @@ impl VfioPciDevice {
                 let (mmap_offset, _) = self.device.get_region_mmap(region.index);
 
                 // Remove region
-                let r = self.vm.make_user_memory_region(
+                let r = self.vm.lock().unwrap().make_user_memory_region(
```

Comment:
Use `let vm = vm.lock().unwrap()` at the beginning. Please check all other places where the same pattern can be used.

Reply:
Checked and fixed all such occurrences.  I only fixed instances where moving the initialization to top, would improve performance.  I left all single invocation as is.